spec/system/voting_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Voting', type: :system, js: true do
+  fab!(:user) { Fabricate(:user) }
+  fab!(:admin) { Fabricate(:admin) }
+  fab!(:category1) { Fabricate(:category) }
+  fab!(:category2) { Fabricate(:category) }
+  fab!(:topic1) { Fabricate(:topic, category: category1) }
+  fab!(:topic2) { Fabricate(:topic, category: category1) }
+  fab!(:topic3) { Fabricate(:topic, category: category2) }
+  fab!(:post1) { Fabricate(:post, topic: topic1) }
+  fab!(:post2) { Fabricate(:post, topic: topic2) }
+  fab!(:category_page) { PageObjects::Pages::Category.new }
+  fab!(:topic_page) { PageObjects::Pages::Topic.new }
+  fab!(:user_page) { PageObjects::Pages::User.new }
+  fab!(:admin_page) { PageObjects::Pages::Admin.new }
+
+  before do
+    SiteSetting.voting_enabled = false
+
+    admin.activate
+    user.activate
+
+    sign_in(admin)
+  end
+
+  it 'enables voting in category topics and votes' do
+    # on the fence about assertions within page objects
+    # but they do provide some readable "assertions"
+    # rather than
+    category_page.visit(category1)
+    expect(category_page).to have_no_css(category_page.votes)
+
+    # enables voting
+    admin_page
+      .visit_filtered_plugin_setting('voting%20enabled')
+      .toggle_setting('Allow users to vote on topics?')
+
+    category_page
+      .visit_settings(category1)
+      .toggle_setting('Allow users to vote on topics in this category')
+      .save_settings
+      .back_to_category
```

Comment:
@jjaffeux Like these ones, nice to chain I think.

Actions that lead us to another page should probably not return self.

Reply:
Not against the idea, but it might be hard to know just from reading it 🤔

Reply:
For sure. I was thinking to let the test runtime 'speak' and have the writer realise that the page moved on.